pkg/memo/header.go

```diff
@@ -0,0 +1,137 @@
+package memo
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	zetamath "github.com/zeta-chain/node/pkg/math"
+)
+
+const (
+	// Identifier is the ASCII code of 'Z' (0x5A)
+	Identifier byte = 0x5A
+
+	// HeaderSize is the size of the memo header: [identifier + ctrlByte1+ ctrlByte2 + dataFlags]
+	HeaderSize = 4
+
+	// MaskVersion is the mask for the version bits(upper 4 bits)
+	MaskVersion byte = 0b11110000
+
+	// MaskEncodingFormat is the mask for the encoding format bits(lower 4 bits)
+	MaskEncodingFormat byte = 0b00001111
+
+	// MaskOpCode is the mask for the operation code bits(upper 4 bits)
+	MaskOpCode byte = 0b11110000
+
+	// MaskCtrlReserved is the mask for reserved control bits (lower 4 bits)
+	MaskCtrlReserved byte = 0b00001111
+)
+
+// Enum for non-EVM chain inbound operation code (4 bits)
+const (
+	OpCodeDeposit        uint8 = 0b0000 // operation 'deposit'
+	OpCodeDepositAndCall uint8 = 0b0001 // operation 'deposit_and_call'
+	OpCodeCall           uint8 = 0b0010 // operation 'call'
+	OpCodeInvalid        uint8 = 0b0011 // invalid operation code
+)
+
+// Header represent the memo header
+type Header struct {
+	// Version is the memo Version
+	Version uint8
+
+	// EncodingFormat is the memo encoding format
+	EncodingFormat uint8
+
+	// OpCode is the inbound operation code
+	OpCode uint8
```

Comment:
Consider having a dedicated type `type OpCode uint8`, same for `type EncodingFormat`

Reply:
https://github.com/zeta-chain/node/pull/2987/commits/d022be1053ebba9c11d5d3d5c51755b3705b4c03